docs/source/extensions/accelerator.rst

```diff
@@ -1,51 +1,111 @@
-    from pytorch_lightning.plugins import NativeMixedPrecisionPlugin
-    from pytorch_lightning.strategies import DDPStrategy
+We expose Accelerators and Strategies mainly for expert users who want to extend Lightning to work with new
+hardware and distributed training or clusters.
 
-    accelerator = GPUAccelerator()
-    precision_plugin = NativeMixedPrecisionPlugin(precision=16, device="cuda")
-    training_strategy = DDPStrategy(accelerator=accelerator, precision_plugin=precision_plugin)
-    trainer = Trainer(strategy=training_strategy, devices=2)
 
+----------
+
+Create a Custom Accelerator
+---------------------------
+
+Here is how you create a new Accelerator.
+Let's pretend we want to integrate the fictional XPU accelerator and we have access to its hardware through a library
+``xpulib``.
+
+.. code-block:: python
+
+    import xpulib
 
-We expose Accelerators and Plugins mainly for expert users who want to extend Lightning to work with new
-hardware and distributed training or clusters.
 
+    class XPUAccelerator(Accelerator):
+        """Experimental support for XPU, optimized for large-scale machine learning."""
 
-.. image:: ../_static/images/accelerator/overview.svg
+        @staticmethod
+        def parse_devices(devices: Any) -> Any:
+            # Put parsing logic here how devices can be passed into the Trainer
+            # via the `devices` argument
+            return devices
 
+        @staticmethod
+        def get_parallel_devices(devices: Any) -> Any:
+            # Here, convert the device indices to actual device objects
+            return [torch.device("xpu", idx) for idx in devices]
+
+        @staticmethod
+        def auto_device_count() -> int:
+            # Return a value for auto-device selection when `Trainer(devices="auto")`
+            return xpulib.available_devices()
+
+        @staticmethod
+        def is_available() -> bool:
+            return xpulib.is_available()
+
+        def get_device_stats(self, device: Union[str, torch.device]) -> Dict[str, Any]:
+            # Return optional device statistics for loggers
+            return {}
+
+
+Finally, add the XPUAccelerator to the Trainer:
+
+.. code-block:: python
+
+    from pytorch_lightning import Trainer
+
+    accelerator = XPUAccelerator()
+    trainer = Trainer(accelerator=accelerator, devices=2)
+
+
+:doc:`Learn more about Strategies <../extensions/strategy>` and how they interact with the Accelerator.
 
-.. warning:: The Accelerator API is in beta and subject to change.
-    For help setting up custom plugins/accelerators, please reach out to us at **support@pytorchlightning.ai**
 
 ----------
 
+Registering Accelerators
+------------------------
+
+If you wish to switch to a custom accelerator from the CLI without code changes, you can use the :class:`~pytorch_lightning.accelerators.registry.AcceleratorRegistry` to register your new accelerator under a shorthand name like so:
```

Comment:
User could include `@classmethod def register_accelerators` in the customized accelerator class to register their accelerator too. I think suggesting user add register function in the customized class, and provide this as additional way is more clear?

Are we gonna update this one as well? https://pytorch-lightning.readthedocs.io/en/latest/advanced/strategy_registry.html

Reply:
Yes, we may want to update the strategy registry too. Leaving it for a separate PR because it's a separate page atm.